src/test/java/itest/MessagingServerIT.java

```
@@ -0,0 +1,81 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.json.JsonObject;
+import itest.bases.StandardSelfTest;
+import itest.util.Utils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+public class MessagingServerIT extends StandardSelfTest {
+
+    public static final String DEPRECATED_COMMAND_PATH = "/api/v1/command";
+
+    public static CompletableFuture<JsonObject> sendMessage(String command, String... args)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        CompletableFuture<JsonObject> future = new CompletableFuture<>();
+
+        Utils.HTTP_CLIENT.webSocket(
+                DEPRECATED_COMMAND_PATH,
+                ar -> {
+                    if (ar.failed()) {
+                        future.completeExceptionally(ar.cause());
+                        return;
+                    }
+                });
+
+        return future;
+    }
+
+    @Test
+    public void shouldRejectDeprecatedCommandPathWith410StatusCode() {
+        try {
+            sendMessage("ping").get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            String errorMessage = "Websocket connection attempt returned HTTP status code 410";
+            MatcherAssert.assertThat(e.getCause().getMessage(), Matchers.equalTo(errorMessage));
```

When you're writing a test that expects an exception to occur, it's best to write it like this:

```
SomeException ex = Assertions.assertThrows(SomeException.class, () -> doSomething());
MatcherAssert.assertThat(ex, Matchers.hasSomeProperties());
```

This way, the test will immediately fail if the expected exception is *not* thrown. Simply using a `try/catch` the typical/intuitive way means that if no exception is thrown, the test will skip the `catch` clause and silently pass, never hitting any assertions along the way, possibly resulting in a false positive.